Docker/StorageApi/Writer.php

```diff
@@ -422,6 +445,33 @@ protected function uploadTable($source, $config = array())
                     // ignore
                 }
             }
+
+            if (self::modifyPrimaryKeyDecider($this->features, $tableInfo, $config)) {
+                $this->getLogger()->warn("Modifying primary key of table {$tableInfo["id"]}.");
+                $failed = false;
+                // modify primary key
+                try {
+                    $this->client->removeTablePrimaryKey($tableInfo["id"]);
+                } catch (\Exception $e) {
+                    // warn and go on
+                    $this->getLogger()->warn("Error deleting primary key of table {$tableInfo["id"]}:" . $e->getMessage());
+                    $failed = true;
+                }
+                if (!$failed) {
+                    try {
+                        $configPK = self::normalizePrimaryKey($config["primary_key"]);
+                        if (count($configPK)) {
+                            $this->client->createTablePrimaryKey($tableInfo["id"], $configPK);
```

Comment:
tohle vlastne znamena, ze kdyz bude mit tabulka pk a config pk je mit nebude, tak se smazou, je to fakt zadouci?